File: create-pwa-icons.js
#!/usr/bin/env node
const canvas = require('canvas');
const fs = require('fs');
const path = require('path');

const Color = {
  primary: '#28655c',    // Theme color
  white: '#ffffff'
};

function createIcon(size, isMaskable = false) {
  const c = canvas.createCanvas(size, size);
  const ctx = c.getContext('2d');
  
  // Background
  ctx.fillStyle = isMaskable ? Color.white : Color.primary;
  ctx.fillRect(0, 0, size, size);
  
  if (isMaskable) {
    // Draw circle in the center
    ctx.fillStyle = Color.primary;
    ctx.beginPath();
    ctx.arc(size / 2, size / 2, size / 3, 0, Math.PI * 2);
    ctx.fill();
  }
  
  // Text - "AM"
  ctx.fillStyle = Color.white;
  ctx.font = `bold ${size / 3}px Arial`;
  ctx.textAlign = 'center';
  ctx.textBaseline = 'middle';
  ctx.fillText('AM', size / 2, size / 2);
  
  return c;
}

const iconsDir = path.join(__dirname, 'public', 'icons');

// Ensure directory exists
if (!fs.existsSync(iconsDir)) {
  fs.mkdirSync(iconsDir, { recursive: true });
}

// Create icons
const icons = [
  { size: 192, file: 'icon-192.png', maskable: false },
  { size: 192, file: 'icon-192-maskable.png', maskable: true },
  { size: 512, file: 'icon-512.png', maskable: false },
  { size: 512, file: 'icon-512-maskable.png', maskable: true }
];

icons.forEach(({ size, file, maskable }) => {
  const c = createIcon(size, maskable);
  const filePath = path.join(iconsDir, file);
  const buffer = c.toBuffer('image/png');
  fs.writeFileSync(filePath, buffer);
  console.log(`✅ Created ${file} (${size}x${size}${maskable ? ' maskable' : ''})`);
});

console.log('\n🎉 All PWA icons created in public/icons/!');

File: create-pwa-icons.py
#!/usr/bin/env python3
from PIL import Image, ImageDraw

# Couleur theme: #28655c
color = (40, 101, 92)
text_color = (255, 255, 255)

icons_path = '/Users/user/Desktop/projects/personnel-projects/malikina/public/icons'

# Icon 192x192
img = Image.new('RGB', (192, 192), color)
draw = ImageDraw.Draw(img)
draw.text((96, 96), "AM", fill=text_color, anchor="mm")
img.save(f'{icons_path}/icon-192.png')
print(f"✅ Created icon-192.png")

# Icon 192x192 maskable
img = Image.new('RGB', (192, 192), (255, 255, 255))
draw = ImageDraw.Draw(img)
draw.ellipse([40, 40, 152, 152], fill=color)
draw.text((96, 96), "AM", fill=text_color, anchor="mm")
img.save(f'{icons_path}/icon-192-maskable.png')
print(f"✅ Created icon-192-maskable.png")

# Icon 512x512
img = Image.new('RGB', (512, 512), color)
draw = ImageDraw.Draw(img)
draw.text((256, 256), "AM", fill=text_color, anchor="mm")
img.save(f'{icons_path}/icon-512.png')
print(f"✅ Created icon-512.png")

# Icon 512x512 maskable
img = Image.new('RGB', (512, 512), (255, 255, 255))
draw = ImageDraw.Draw(img)
draw.ellipse([100, 100, 412, 412], fill=color)
draw.text((256, 256), "AM", fill=text_color, anchor="mm")
img.save(f'{icons_path}/icon-512-maskable.png')
print(f"✅ Created icon-512-maskable.png")

print(f"\n✅ All PWA icons created in {icons_path}!")
